factorial.js

```diff
@@ -1,24 +1,47 @@
+
+/**
+ * Calculates the factorial of a given non-negative integer using an iterative approach.
+ *
+ * @param {number} n - The non-negative integer for which the factorial needs to be calculated.
+ * @returns {number|string} - The factorial of the given number, or an error message if the input is negative.
```

Comment:
Give me detailed doc string as per the swagger API formating. Also I dont need inline comments